game/README.md

```diff
@@ -6,7 +6,263 @@
+│   ├── pkg						# provider and consumer
+│   └── tests
+└── pkg
+    ├── consumer			# public consumer
+    │   ├── game
+    │   └── gate
+    └── pojo
+```
+
+Process of initiating HTTP service
+
+<img src="http://cdn.cjpa.top/cdnimages/image-20210424095907886.png" alt="image-20210424095907886" style="zoom: 33%;" />
+
+
+
+From the perspective of consumer and provider, the process of initiating a call is like this
+
+<img src="http://cdn.cjpa.top/cdnimages/image-20210424100148028.png" alt="image-20210424100148028" style="zoom:33%;" />
+
+Game provides the basketball server and gate provides the HTTP server.
+
+### game module
+
+#### server side
+
+The server provides three services, message, online and offline. The code is as follows. The specific implementation can be in  game/go-server-game/pkg/provider.go see in.
+
+```go
+type BasketballService struct{}
+
+func (p *BasketballService) Online(...) (...) {
+    ...
+}
+
+func (p *BasketballService) Offline(...) (...) {
+    ...
+}
+
+func (p *BasketballService) Message(...) (...) {
+    ...
+}
+
+func (p *BasketballService) Reference() string {
+    return "gameProvider.basketballService"
+}
+```
+
+##### configuration file
+
+Register the service in the configuration file, where gameProvider.basketballService Same as declared in the reference method.
+
+```yml
+services:
+  "gameProvider.basketballService":
+    registry: "demoZk"
+    protocol: "dubbo"
+    interface: "org.apache.dubbo.game.BasketballService"
+    loadbalance: "random"
+    warmup: "100"
+    cluster: "failover"
+    methods:
+      - name: "Online"
+        retries: 0
+      - name: "Offline"
+        retries: 0
+      - name: "Message"
+        retries: 
+```
+
+#### consumer side
+
+The consumer of basketball is mainly used to be called by gate, so the main code is put in game/pkg/consumer/gate/basketball.go,  The consumer code is as follows.
+
+```go
+type BasketballService struct {
+    Send func(ctx context.Context, uid string, data string) (*pojo.Result, error)
+}
+
+func (p *BasketballService) Reference() string {
+    return "gateConsumer.basketballService"
+}
+```
+
+In basketball, you only need to instantiate a consumer variable.
+
+```go
+var gateBasketball = new(gate.BasketballService)
+```
+
+Then register to Dubbo go in main.
+
+```go
+config.SetConsumerService(gateBasketball)
+```
+
+##### configuration file
+
+```yml
+references:
+  "gateConsumer.basketballService":
+    registry: "demoZk"
+    protocol : "dubbo"
+    interface : "org.apache.dubbo.gate.BasketballService"
+    cluster: "failover"
+    methods:
+      - name: "Send"
+        retries: 0
+```
+
+Because the consumer of game needs to call the provider of gate, the string returned by the reference method is gateConsumer.basketballService  , in the configuration file gateConsumer.basketballService  And game/pkg/consumer/gate/basketball.go  The value of intreface should be consistent with that of the provider of gate.
```

Comment:
Because the consumer of game needs to call the provider of gate, the string returned by the reference method is gateConsumer.basketballService , in the configuration file gateConsumer.basketballService  And 'game/pkg/consumer/gate/basketball.go'. The value of interface should be consistent with that of the provider of gate.